src/libcore/alloc/layout.rs

```diff
@@ -15,14 +15,13 @@ const fn size_align<T>() -> (usize, usize) {
 /// An instance of `Layout` describes a particular layout of memory.
 /// You build a `Layout` up as an input to give to an allocator.
 ///
-/// All layouts have an associated non-negative size and a
-/// power-of-two alignment.
+/// All layouts have an associated size and a power-of-two alignment.
 ///
-/// (Note however that layouts are *not* required to have positive
-/// size, even though many allocators require that all memory
-/// requests have positive size. A caller to the `AllocRef::alloc`
-/// method must either ensure that conditions like this are met, or
-/// use specific allocators with looser requirements.)
+/// (Note however that layouts are *not* required to have a non-size,
+/// even though `GlobalAlloc` require that all memory requests
+/// are non-zero in size. A caller must either ensure that conditions
+/// like this are met, or use specific allocators with looser
+/// requirements. `AllocRef`)
```

Comment:
(Mostly just a grammar nitpick)
```suggestion
/// (Note that layouts are *not* required to have non-zero size,
/// even though `GlobalAlloc` requires that all memory requests
/// be non-zero in size. A caller must either ensure that conditions
/// like this are met, use specific allocators with looser
/// requirements, or use the more lenient `AllocRef` interface.)
```

Reply:
Can't tell why "requirements. `AllocRef`)" made it into the commit :smile: